### PR TITLE
Scheduler Performance Improvements

### PR DESCRIPTION
**Pull request by abhay-agarwal**
Added fix for #117 Dangling Executor for 15 minutes after slave restart

### CODE REVIEW

**Comment by elingg** [comment]
hey @abhay-agarwal, could you separate the kill task null check into a separate PR from the scheduler changes please? Also, since we are no longer launching JN in NameNodeExecutor it will always be null, can you remove it from that executor instead of having the null check

**Comment by elingg** [comment]
Thanks for breaking this down @abhay-agarwal, let me see if I can fix this in a simpler way with my reconciliation changes. I'll let you know!

**Comment by elingg** [comment]
We should be able to close this PR since I have done: https://github.com/mesosphere/hdfs/pull/120 most likely.

File: src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java [hunk 1]
```diff
@@ -100,9 +100,11 @@ public void killTask(ExecutorDriver driver, TaskID taskId) {
   public void shutdown(ExecutorDriver d) {
     // TODO(elingg) let's shut down the driver more gracefully
     log.info("Executor asked to shutdown");
-    killTask(d, journalNodeTask.taskInfo.getTaskId());
     killTask(d, nameNodeTask.taskInfo.getTaskId());
     killTask(d, zkfcNodeTask.taskInfo.getTaskId());
+    if (journalNodeTask != null) {
```
Comment (adam-mesos) [inline src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java]: Do we need to check if nameNodeTask != null or zkfcNodeTask != null too?

**Comment by elingg** [comment]
Hi @adam-mesos, please do not review this. We will most likely be closing this and instead committing #120.

**Pull request closed without merging**